crypto/ec/ec_pmeth.c

```diff
@@ -121,8 +125,23 @@ static int pkey_ec_sign(EVP_PKEY_CTX *ctx, unsigned char *sig, size_t *siglen,
     }
 
     type = (dctx->md != NULL) ? EVP_MD_type(dctx->md) : NID_sha1;
-
-    ret = ECDSA_sign(type, tbs, tbslen, sig, &sltmp, ec);
+	
+    switch (dctx->nonce_type) {
```

Comment:
@slontis nid is set in dctx->nonce_type, if use random K (legacy behavior), it will call ECDSA_sign(), which ignores type parameter,
If use deterministic K, it will call ec->meth->sign directly and pass type parameter to it.
This method works well, except that I don't know where you want me to call a ctrl method?

Reply:
arent you already doing that to set the nonce_type?
I am going to sleep soon.

Reply:
@slontis I feel really confused about your words...Maybe we are not talking about the same thing?
We may talk about it tomorrow, have a good dream!

Reply:
ok.. I will look again tomorrow.

Reply:
@slontis Have you reviewed the code these days? Is there anything else that I need to change?